### DISCUSSION TITLE
Fall back to epoll when uring is not usable

### DISCUSSION THREAD

**Discussion by rezamahdi** [libuv/libuv#4031] [General]
After adding support for io_uring, which I personally waited for it many times, it noted that when the kernel doesn't have the feature, libuv will dramatically fall back to thread pool.

io_uring is relatively get matured but many distributions don't use most recent LTS kernels yet so deprecating epoll is not a good decision yet...

Libuv really discarded the epoll completely??

**Comment by bnoordhuis** [comment]
No. io_uring is used for operations that previously went through the thread pool, no epoll involved.
